File: benchmarks/javascript/sort_bench.js
const RUNS = 100;
const SIZE = 100000;

// Seeded random for reproducibility
function seededRand(seed) {
    const arr = new Int32Array(SIZE);
    let s = seed;
    for (let i = 0; i < SIZE; i++) {
        s = (s * 1664525 + 1013904223) & 0x7fffffff;
        arr[i] = s % 1000000;
    }
    return arr;
}

const original = seededRand(42);
const arr = new Int32Array(SIZE);

// warmup
arr.set(original);
arr.sort((a, b) => a - b);

let sink = 0;
const start = performance.now();
for (let r = 0; r < RUNS; r++) {
    arr.set(original);
    arr.sort((a, b) => a - b);
    sink += arr[0];
}
const end = performance.now();

console.log(`Integer Sort JS mean: ${((end - start) / RUNS).toFixed(2)} ms (sink: ${sink})`);

File: benchmarks/native/sort_bench.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <time.h>

#define SIZE 100000

int original[SIZE];
int arr[SIZE];

int cmp(const void *a, const void *b) {
    return (*(int*)a - *(int*)b);
}

int main() {
    srand(42);
    for (int i = 0; i < SIZE; i++)
        original[i] = rand() % 1000000;

    int RUNS = 100;
    struct timespec start, end;
    volatile int sink = 0;

    clock_gettime(CLOCK_MONOTONIC, &start);
    for (int r = 0; r < RUNS; r++) {
        memcpy(arr, original, SIZE * sizeof(int));
        qsort(arr, SIZE, sizeof(int), cmp);
        sink += arr[0];
    }
    clock_gettime(CLOCK_MONOTONIC, &end);

    double ms = ((end.tv_sec - start.tv_sec)*1e9 + (end.tv_nsec - start.tv_nsec)) / 1e6 / RUNS;
    printf("Integer Sort mean: %.2f ms (sink: %d)\n", ms, sink);
    return 0;
}
